lib/warden/strategies/azure_ad.rb

```diff
@@ -2,27 +2,14 @@ module Warden
   module Strategies
     class AzureAd < Warden::Strategies::Base
       def authenticate!
-        omniauth_user = env['omniauth.auth']
+        info = env.dig('omniauth.auth', 'info')
 
-        if omniauth_user
-          info = env['omniauth.auth']['info']
-
-          success! find_and_update_user(info)
+        if info && (user = User.authenticate!(info))
```

Comment:
Perhaps move the instantiation of user out of the statement here

Reply:
will fix in later pr